[incubator/sentry-kubernetes] add missing clusterrole and clusterrolebinding

jcscottiii opened the pull request:
This PR adds the missing clusterrole and clusterrolebinding needed for the sentry-kubernetes chart.

It was part of the original PR https://github.com/kubernetes/charts/pull/2708 but wasn't formatted right.
You can see more about it [here](https://github.com/getsentry/sentry-kubernetes/issues/4) about why you need the role and binding.

Commit:
[incubator/sentry-kubernetes] add missing clusterrole and clusterrolebinding

jcscottiii commented:
~I added my other email address to my account. Do I need to do anything to fix the `missing Github user` error?~

EDIT: it's working now.

unguiculus commented on incubator/sentry-kubernetes/templates/clusterrolebinding.yaml:
```diff
@@ -0,0 +1,15 @@
+  name: {{ template "sentry-kubernetes.fullname" . }}
+subjects:
+  - kind: ServiceAccount
+    name: {{ template "sentry-kubernetes.fullname" . }}
```
`{{ template "sentry-kubernetes.serviceAccountName" . }}`

jcscottiii replied:
whoops!! @unguiculus  thanks for that catch. (overlooked from copying over from closed PR)
fixed it now.

jcscottiii replied:
forgot to mention that it just happened to work for me because `{{ template "sentry-kubernetes.fullname" . }}` and `{{ template "sentry-kubernetes.serviceAccountName" . }}` evaluated to the same thing. but yeah, we should definitely be using `{{ template "sentry-kubernetes.serviceAccountName" . }}`

unguiculus replied:
Please fix it in the deployment as well.

jcscottiii replied:
@unguiculus great catch. fixed it!

unguiculus reviewed (changes requested):
Thanks for this. I noticed that the service acount is not correctly referenced in the deployment. It should also use `{{ template "sentry-kubernetes.serviceAccountName" . }}`. So, while you're at it, could please also fix this. Thanks.

See https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

cc @gianrubio

Commit:
[sentry-kubernetes] use correct helper for SA

previous commit introduced clusterrole and clusterrolebinding
but the subject for the SA was not using the right template helper.
this commit fixes that.

gianrubio commented:
ohhh nice catch @jcscottiii, I forgot to add the rbac rules on my git commit, thanks for your PR!

jdumars commented:
/ok-to-test

Commit:
[sentry-kubernetes] fix SA variable in deployment

jcscottiii commented:
just added a new commit per https://github.com/kubernetes/charts/pull/3942#discussion_r171934178 should be good to go :)

unguiculus commented on incubator/sentry-kubernetes/templates/deployment.yaml:
```diff
@@ -34,4 +34,4 @@ spec:
 {{ toYaml .Values.tolerations | indent 8 }}
     {{- end }}
-      serviceAccountName: {{ if .Values.rbac.create }}{{ template "sentry-kubernetes.fullname" . }}{{ else }}"{{ .Values.rbac.serviceAccountName }}"{{ end }}
+      serviceAccountName: {{ if .Values.rbac.create }}{{ template "sentry-kubernetes.serviceAccountName" . }}{{ else }}"{{ .Values.rbac.serviceAccountName }}"{{ end }}
```
Just this:
`serviceAccountName: {{ template "sentry-kubernetes.serviceAccountName" . }}`

jcscottiii replied:
fixed! thanks again

jcscottiii replied:
@unguiculus any other feedback?

jcscottiii committed:
[sentry-kubernetes] remove redundant conditional

unguiculus commented:
/lgtm
